File: dotnet/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/Services/Concretes/EmailNotificationService.cs
using FactoryDesignPattern.NotificationAPI.Models;
using FactoryDesignPattern.NotificationAPI.Services.Abstracts;

namespace FactoryDesignPattern.NotificationAPI.Services.Concretes;

public class EmailNotificationService : INotificationService
{
	public string Send(string to, string message)
	{
		string result = $"Sending Email to {to} with message: {message}";
		Console.WriteLine(result);

		return result;
	}
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/FactoryDesignPattern/FactoryDesignPattern.Example1/src/services/EmailNotificationService.ts
import { INotificationService } from './INotificationService';

export class EmailNotificationService implements INotificationService {
  send(to: string, message: string): string {
    const result = `Sending Email to ${to} with message: ${message}`;
    console.log(result);
    return result;
  }
}
